### ISSUE TITLE
'Invalid version string "tap"' in "npm-asset/save-pixels"

### ISSUE THREAD

**Issue by SoulCover** [francoispluchino/composer-asset-plugin#101]
Just stumbled upon your ingenious project and wanted to try it out with an existing Typo3 Flow project. Unfortunately while checking dependencies the following exception is thrown by composer:
````
[UnexpectedValueException]
Could not parse version constraint tap: Invalid version string "tap"
````
Luckily I could find the culprit. I made a new empty directory and just required `npm-asset/save-pixels` (\*). Same exception. Their package.json is [over here](https://github.com/scijs/save-pixels/blob/master/package.json). But now I'm scratching my head and guess the `"scripts": { "test": "tap test/*.js" }` part is the reason.

**Comment by SoulCover** [comment]
`npm info save-pixels` gives me this btw:
````js
{ name: 'save-pixels',
  description: 'Saves an ndarray as an image to a file',
  'dist-tags': { latest: '2.2.0' },
  versions:
   [ '0.0.0',
     '0.0.1',
     '0.1.0',
     '0.1.2',
     '0.2.0',
     '0.2.1',
     '0.3.0',
     '1.0.0',
     '2.0.0',
     '2.1.0',
     '2.2.0' ],
  maintainers: 'mikolalysenko <mikolalysenko@gmail.com>',
  time:
   { modified: '2014-11-12T14:38:45.271Z',
     created: '2013-05-15T03:37:52.235Z',
     '0.0.0': '2013-05-15T03:37:53.518Z',
     '0.0.1': '2013-05-15T17:46:34.406Z',
     '0.1.0': '2013-05-20T17:23:21.928Z',
     '0.1.2': '2013-05-21T17:31:03.302Z',
     '0.2.0': '2013-06-26T17:31:26.758Z',
     '0.2.1': '2013-07-30T18:15:41.889Z',
     '0.3.0': '2013-07-30T21:59:17.483Z',
     '1.0.0': '2014-05-22T07:46:59.054Z',
     '2.0.0': '2014-10-18T21:25:11.851Z',
     '2.1.0': '2014-10-18T21:30:36.662Z',
     '2.2.0': '2014-11-12T14:38:45.271Z' },
  author: 'Mikola Lysenko',
  repository:
   { type: 'git',
     url: 'git://github.com/mikolalysenko/save-pixels.git' },
  homepage: 'https://github.com/mikolalysenko/save-pixels',
  keywords:
   [ 'save',
     'pixel',
     'image',
     'png',
     'ndarray' ],
  bugs: { url: 'https://github.com/mikolalysenko/save-pixels/issues' },
  license: 'MIT',
  readmeFilename: 'README.md',
  users: { philippwiddra: true },
  version: '2.2.0',
  main: 'save-pixels.js',
  directories: { example: 'example', test: 'test' },
  dependencies:
   { contentstream: '^1.0.0',
     'gif-encoder': '~0.4.1',
     'jpeg-js': '0.0.4',
     pngjs: '~0.4.0-alpha',
     through: '^2.3.4' },
  devDependencies:
   { 'get-pixels': '^3.0.1',
     ndarray: '^1.0.0',
     'ndarray-scratch': '^1.1.1',
     tape: '^3.0.1' },
  scripts: { test: 'tap test/*.js' },
  gitHead: 'fda0977a1a5f2797d66f1557ffecc8c2dc759a47',
  dist:
   { shasum: 'beab5d74a4400731b22474b2e477f7626bfd193f',
     tarball: 'http://registry.npmjs.org/save-pixels/-/save-pixels-2.2.0.tgz' } }
````

**Comment by francoispluchino** [comment]
It's a bug, see `https://registry.npmjs.org/save-pixels`.

There is a special version in devDependencies section:

```json
{
    "devDependencies": {
        "tap": "tap test/*.js"
    }
}
```

**Comment by francoispluchino** [comment]
Fixed by dc24e6280386d1e43be50c9459f6ca8f80965a4a.
